subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -698,7 +704,7 @@ void ull_sync_setup(struct ll_scan_set *scan, struct ll_scan_aux_set *aux,
 	/* Remember the peer address.
 	 * NOTE: Peer identity address is copied here when privacy is enable.
 	 */
-	sync->peer_id_addr_type = scan->periodic.adv_addr_type;
+	sync->peer_id_addr_type = scan->periodic.adv_addr_type & 0x01;
```

Comment:
In the host we are doing this translation by
```c
dst->type &= ~ADDR_RESOLVED_BITMASK;
```

Perhaps at some point we should add and use a common macro/function to translate the type. 

Ditto for marking it as resolved